crates/rpc/rpc/src/eth/logs_utils.rs

```diff
@@ -110,15 +110,18 @@ pub(crate) fn log_matches_filter(
     log: &reth_primitives::Log,
     params: &FilteredParams,
 ) -> bool {
-    if params.filter.is_some() &&
-        (!params.filter_block_range(block.number) ||
-            !params.filter_block_hash(block.hash) ||
-            !params.filter_address(&from_primitive_log(log.clone())) ||
-            !params.filter_topics(&from_primitive_log(log.clone())))
-    {
-        return false
+    if params.filter.is_none() {
+        return true
     }
-    true
+
+    if params.filter_block_range(block.number) && params.filter_block_hash(block.hash) {
+        let log = from_primitive_log(log.clone());
+        if params.filter_address(&log) && params.filter_topics(&log) {
```

Comment:
I'm unsure if this is equivalent to the || because block_range and block_hash can be true independently of each other?